### PR TITLE
[async-mqtt] Add new port

### PR DESCRIPTION
**Pull request by redboltz**
If this PR adds a new port, please uncomment and fill out this checklist:

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

END OF NEW PORT CHECKLIST (delete this line)

### CODE REVIEW

**Comment by redboltz** [comment]
@microsoft-github-policy-service agree

File: ports/async-mqtt/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+vcpkg_cmake_install()
+
+vcpkg_cmake_config_fixup(PACKAGE_NAME async_mqtt_iface CONFIG_PATH lib/cmake/async_mqtt_iface)
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug" "${CURRENT_PACKAGES_DIR}/lib")
+
+vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")
+
+file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```
Comment (FrankXie05) [inline ports/async-mqtt/portfile.cmake:27]:
```suggestion
```
Just need function `vcpkg_install_copyright`.

File: ports/async-mqtt/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,27 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO redboltz/async_mqtt
+    REF 1.0.1
```
Comment (FrankXie05) [inline ports/async-mqtt/portfile.cmake:4]:
```suggestion
    REF "${VERSION}"
```
We can use the new function.

File: ports/async-mqtt/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    -DASYNC_MQTT_BUILD_TOOLS=OFF
+    -DASYNC_MQTT_BUILD_EXAMPLES=OFF
+    -DASYNC_MQTT_BUILD_UNIT_TESTS=OFF
+    -DASYNC_MQTT_BUILD_SYSTEM_TESTS=OFF
+    -DCMAKE_DISABLE_FIND_PACKAGE_Doxygen=ON
+)
+
+vcpkg_cmake_install()
+
+vcpkg_cmake_config_fixup(PACKAGE_NAME async_mqtt_iface CONFIG_PATH lib/cmake/async_mqtt_iface)
```
Comment (FrankXie05) [inline ports/async-mqtt/portfile.cmake:21]:
```suggestion
vcpkg_cmake_config_fixup(PACKAGE_NAME async_mqtt_iface CONFIG_PATH "lib/cmake/async_mqtt_iface")
```

File: ports/async-mqtt/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug" "${CURRENT_PACKAGES_DIR}/lib")
```
Comment (FrankXie05) [inline ports/async-mqtt/portfile.cmake:23]: Is this port a `header-only` or only `dynamic`.?

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

**Comment by FrankXie05** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". 
          That way, I can be aware that you've responded since you can't modify the tags.

File: ports/async-mqtt/portfile.cmake (hunk 4 above)
Comment (redboltz) [reply ports/async-mqtt/portfile.cmake:23]: This portis a header-only.

Comment (redboltz) [reply ports/async-mqtt/portfile.cmake:23]: Should these two lines be removed if the port is a header-only library?

File: ports/async-mqtt/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,27 @@
+vcpkg_from_github(
```
Comment (FrankXie05) [inline ports/async-mqtt/portfile.cmake:1]:
```suggestion
#header-only library

vcpkg_from_github(
```

File: ports/async-mqtt/portfile.cmake (hunk 4 above)
Comment (FrankXie05) [reply ports/async-mqtt/portfile.cmake:23]: No, keep two lines. We don't need empty folders.

File: ports/async-mqtt/portfile.cmake [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS
+    -DASYNC_MQTT_BUILD_TOOLS=OFF
+    -DASYNC_MQTT_BUILD_EXAMPLES=OFF
+    -DASYNC_MQTT_BUILD_UNIT_TESTS=OFF
+    -DASYNC_MQTT_BUILD_SYSTEM_TESTS=OFF
+    -DCMAKE_DISABLE_FIND_PACKAGE_Doxygen=ON
```
Comment (dg0yt) [inline ports/async-mqtt/portfile.cmake:16]: Please indent the actual options:
~~~
    OPTIONS
        -DASYNC_MQTT_BUILD_TOOLS=OFF
        -DASYNC_MQTT_BUILD_EXAMPLES=OFF
        -DASYNC_MQTT_BUILD_UNIT_TESTS=OFF
        -DASYNC_MQTT_BUILD_SYSTEM_TESTS=OFF
        -DCMAKE_DISABLE_FIND_PACKAGE_Doxygen=ON
~~~

**Comment by redboltz** [comment]
I think that all changes request is reflected.

File: ports/async-mqtt/portfile.cmake (hunk 5 above)
Comment (redboltz) [reply ports/async-mqtt/portfile.cmake:3]: I looked over this comment. I just reflected it.

**Comment by redboltz** [comment]
https://github.com/microsoft/vcpkg/pull/31902#discussion_r1224148906 is fixed too.
So now ready to review.

**Review by dg0yt — APPROVED** [review]
Just from looking at it in GH, only one nit-pick left.

The port is submitted from the owner, and it is not yet a project with many GH stars, so vcpkg maintainers must decide if they want the port under this name or under `redboltz-async-mqtt`. (There is at least a JS project under the same name. https://github.com/search?q=async-mqtt&type=repositories)

**Comment by redboltz** [comment]
> Just from looking at it in GH, only one nit-pick left.
> 
> The port is submitted from the owner, and it is not yet a project with many GH stars, so vcpkg maintainers must decide if they want the port under this name or under `redboltz-async-mqtt`. (There is at least a JS project under the same name. https://github.com/search?q=async-mqtt&type=repositories)

Thank you for the comment. I fixed that removing debug directory.
Let me explain some background. I am the author both 
https://github.com/redboltz/async_mqtt and
https://github.com/redboltz/mqtt_cpp .
mqtt_cpp has been registered in vcpkg. Some contributor did it.
async_mqtt is successor project of mqtt_cpp. It coveres all asynchronous functionality of mqtt_cpp, but using more elegant way.

After I released async_mqtt, vcpkg support is requested by user https://github.com/redboltz/async_mqtt/issues/16
So I createed this PR. I accept async-mqtt, redboltz-async-mqtt, or perhaps async-mqtt-cpp.
(The name of https://github.com/redboltz/async_mqtt is not changed.)

**Review by FrankXie05 — DISMISSED** [review]

**Comment by FrankXie05** [comment]
@redboltz 
It seems that when generating the cmake configuration file, a parenthesis is missing.

`async_mqtt_ifaceConfig.cmake`:
```
set(Boost_USE_MULTITHREADED ON)
include(CMakeFindDependencyMacro)
find_dependency((Threads)
find_dependency((Boost 1.81.0 COMPONENTS )
...
...
```
```
CMake Error at E:/usage-test/vcpkg/installed/x64-windows/share/async_mqtt_iface/async_mqtt_ifaceConfig.cmake:3:
  Parse error.  Function missing ending ")".  End of file reached.
```

**Comment by dg0yt** [comment]
> find_dependency((Boost 1.81.0 COMPONENTS )

There is a double `(`.

**Comment by FrankXie05** [comment]
```
find_dependency((Threads)
```

**Comment by FrankXie05** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". 
          That way, I can be aware that you've responded since you can't modify the tags.

**Comment by redboltz** [comment]
I fixed async_mqtt cmake config issue and then released it as 1.0.2.
Then I updated this PR.

**Comment by redboltz** [comment]
All checks have passed, now ready for review.

**Comment by FrankXie05** [comment]
The usage has been tested successfully locally.
```
async-mqtt provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(async_mqtt_iface CONFIG REQUIRED)
    target_link_libraries(main PRIVATE async_mqtt_iface::async_mqtt_iface)
```

**Review by FrankXie05 — APPROVED** [review]

**Pull request merged**